usaspending_api/etl/elasticsearch_loader_helpers/delete_data.py

```diff
@@ -142,46 +186,92 @@ def delete_docs_by_unique_key(client: Elasticsearch, key: str, value_list: list,
     return deleted
 
 
-def _lookup_deleted_award_ids(client: Elasticsearch, id_list: list, config: dict, index: Optional[str] = None) -> list:
-    """Lookup deleted transactions to derive parent awards to be deleted
+def _is_allowed_key_field_type(client: Elasticsearch, key_field: str, index: str) -> bool:
+    """Return ``True`` if the given field's mapping in the given index is in our allowed list of ES types
+    compatible with term(s) queries
 
-    This fetches a list of all unique award keys compiled from the ``ES_AWARDS_UNIQUE_KEY_FIELD`` field of
-    any document in the transaction index that matches the query, which looks up deleted transaction ES
-    documents by their ``ES_TRANSACTIONS_UNIQUE_KEY_FIELD`` field.
+    This is mainly to prevent use of ``text`` fields in terms queries, which give bad results because Elasticsearch
+    changes the values of text fields during analysis.
+    """
+    if key_field == "_id":
+        # Special case. It is a reserved field, without a type, but can effectively be treated as a keyword field
+        return True
 
-    Args:
-        client (Elasticsearch): elasticsearch-dsl client for making calls to an ES cluster
-        id_list (list): A list of dictionaries, each having two keys, in this format::
+    # Get true index name from alias, if provided an alias
+    response = client.indices.get(index)
+    aliased_index_name = list(response.keys())[0]
+    es_field_type = Mapping().from_es(using=client, index=aliased_index_name).resolve_field(key_field)
+    # This is the allowed types whitelist. More can be added as-needed if compatible with terms(s) queries.
+    if es_field_type and es_field_type.name in ["keyword", "integer"]:
+        return True
+    return False
 
-            id_list = [
-                {key:'<value_of_col>', col:'<unique_key_col_name>'},
-                {key:'<value_of_col>', col:'<unique_key_col_name>'},
-                ...,
-            ]
 
-        config (dict): collection of key-value pairs that encapsulates runtime arguments for this ES management task
+def _lookup_deleted_award_keys(
+    client: Elasticsearch,
+    lookup_key: str,
+    value_list: list,
+    config: dict,
+    index: Optional[str] = None,
+    lookup_chunk_size=50000,
```

Comment:
Does the lookup chunk size need to be variable? Seems like it adds unnecessary complexity from the conditional checks below for only a potentially minuscule value

Reply:
At some point I thought I saw transactions having a query size (results window) of only 10,000 and awards of 50,000. And if not, that seems a possible case where many of the indices we have may have different max_result_window settings.

This also helps us test boundaries and chunks-processing in unit test with small datasets (e.g. setting `lookup_chunk_size=1`).

The conditionals would should still be there, as checks, to compare max_result_window (query size) to the chunk size. This is just having it compared to a variable with a default, rather than a hard-coded `50000` value.

Reply:
> The conditionals would should still be there, as checks, to compare max_result_window (query size) to the chunk size. This is just having it compared to a variable with a default, rather than a hard-coded 50000 value.

Or use `max_result_window` as the chunk size if there is a concern the max result window sizes will be different for the various indexes in the future. I'm not sure what benefit it is to test smaller chunk sizes

Reply:
The testing of smaller chunks is what revealed the hotfix bug that was fixed in https://github.com/fedspendingtransparency/usaspending-api/pull/2931